CRM/Contact/Form/DedupeFind.php

```diff
@@ -51,6 +51,7 @@ public function buildQuickForm() {
     $groupList = array('' => ts('- All Contacts -')) + CRM_Core_PseudoConstant::nestedGroup();
 
     $this->add('select', 'group_id', ts('Select Group'), $groupList, FALSE, array('class' => 'crm-select2 huge'));
+    $this->add('text', 'limit', ts('No of contacts to find matches for '));
```

Comment:
The reason it does the number of contacts not results is that it gives you a way to control the size of the query. If you have a DB of 200,000 contacts it's impossible to know how many contacts you have to do a match.find on to get 1000 results - it could be 100 - it could be 20,000 - so you would need to maybe batch by 100 contacts or something until you get to 1000 results.